### PR TITLE
Tsavorite session reorganization

### PR DESCRIPTION
**Pull request by TedHartMS**
- Rename IFunctions to ISessionFunctions (in prep for StoreFunctions)
- Clean up ITsavoriteSession to SessionFunctionsWrapper, minimizing the per-Context code to just the locking portion
- remove RUMD (Read/Upsert/rMw/Delete) functionality from ClientSession; users should use one of the 4 Contexts (Basic, Lockable, Unsafe, LockableUnsafe) for all such actions. This changed a lot of tests
- Remove the unused RUMDAsync methods
- Speed up some of the more time-consuming UTs

### CODE REVIEW

File: libs/server/AOF/AofProcessor.cs [hunk 1]
```diff
@@ -31,11 +31,13 @@ public sealed unsafe partial class AofProcessor
         /// Session for main store
         /// </summary>
         readonly ClientSession<SpanByte, SpanByte, SpanByte, SpanByteAndMemory, long, MainStoreFunctions> session = null;
```
Comment (badrishc) [inline libs/server/AOF/AofProcessor.cs:33]: do we need to keep `session` and `objectStoreSession` here as fields?

File: libs/storage/Tsavorite/cs/Tsavorite.sln [hunk 2]
```diff
@@ -29,12 +29,12 @@ Global
 		Release|x64 = Release|x64
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{33A732D1-2B58-4FEE-9696-B9483496229F}.Debug|Any CPU.ActiveCfg = Debug|x64
-		{33A732D1-2B58-4FEE-9696-B9483496229F}.Debug|Any CPU.Build.0 = Debug|x64
+		{33A732D1-2B58-4FEE-9696-B9483496229F}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (badrishc) [inline libs/storage/Tsavorite/cs/Tsavorite.sln:32]: is this change to Any CPU intentional?

File: libs/storage/Tsavorite/cs/benchmark/FixedLenYcsbBenchmark.cs [hunk 3]
```diff
@@ -260,34 +215,33 @@ private void RunYcsbSafeContext(int thread_idx)
                 for (long idx = chunk_idx; idx < chunk_idx + YcsbConstants.kChunkSize && !done; ++idx)
                 {
                     if (idx % 512 == 0)
-                        session.CompletePending(false);
+                        bContext.CompletePending(false);
```
Comment (badrishc) [inline libs/storage/Tsavorite/cs/benchmark/FixedLenYcsbBenchmark.cs:218]: does this use of wrapper affect perf numbers?

File: libs/storage/Tsavorite/cs/benchmark/Functions.cs [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 namespace Tsavorite.benchmark
 {
-    public struct Functions : IFunctions<Key, Value, Input, Output, Empty>
+    public struct Functions : ISessionFunctions<Key, Value, Input, Output, Empty>
```
Comment (badrishc) [inline libs/storage/Tsavorite/cs/benchmark/Functions.cs:9]: rename to `struct SessionFunctions`

**Review by badrishc — APPROVED** [review]

File: libs/server/AOF/AofProcessor.cs (hunk 1 above)
Comment (TedHartMS) [reply libs/server/AOF/AofProcessor.cs:33]: Changing to add XxxContext.Session; this is primarily used for Dispose() and also to get Session.ID and Session.Version

File: libs/storage/Tsavorite/cs/Tsavorite.sln (hunk 2 above)
Comment (TedHartMS) [reply libs/storage/Tsavorite/cs/Tsavorite.sln:32]: Yes; we previously built x64 before an earlier csproj reorganization PR. That was a good change so we now always build AnyCPU, but it revealed that this was broken (it's why we were seeing that "Configuration incorrect" message when loading Tsavorite.sln)

File: libs/storage/Tsavorite/cs/benchmark/FixedLenYcsbBenchmark.cs (hunk 3 above)
Comment (TedHartMS) [reply libs/storage/Tsavorite/cs/benchmark/FixedLenYcsbBenchmark.cs:218]: Other way around--this actually removed a wrapper (before, bContext redirected to ClientSession)

**Pull request merged**